src/linkedList/LinkedList.java

```diff
@@ -66,162 +143,223 @@ public int getLength()
-         }
-         else
-         {
-            return -1;
          }
+         this.length++;
+         return this.length;
+      } else
+      {
+         return 0;
+      }
+   }
+
+   /**
+    * Accessor function to return the Object being stored at the head of the 
+    * list. The Object remains in the list.
+    * @return The Object at the head of the list - null if the list is empty.
+    */
+   public E getFirst()
+   {
+      if (this.isEmpty())
+      {
+         return null;
+      }
+      return this.head.getNodeData();
+   }
+   
+   /**
+    * Accessor function to return the Object being stored at the tail of the 
+    * list. The Object remains in the list.
+    * @return The Object at the tail of the list - null if the list is empty.
+    */
+   public E getLast()
+   {
+      if (this.isEmpty())
+      {
+         return null;
+      }
+      return this.tail.getNodeData();
+   }
+   
+   /**
+    * Modifier function to remove a target entry from the list. This function
+    * requires that the caller pass a target object that is comparable to 
+    * another instance of that object being stored in the list.
+    * @param someTarget A comparable Object that you expect to exist in the 
+    * list.
+    * @return The Object that has been removed from the list - null if the list
+    * is empty or if the target was unmatched.
+    */
+   public E removeEntry(E someTarget)
+   {
+      ListNode prevNode;
+      E foundTarget = null;
+
+      if (this.isEmpty())
+      {
+         return null;
+      }
+
+      /* check the head first as findTargetPrev will return a false positive 
+       * in that case.
+       */
+      if (someTarget.equals(this.getFirst()))
+      {
+         foundTarget = (E)this.getFirst();
+         this.head = this.head.getNodeNext();
+         this.length--;
       }
       else
       {
-         return 0;
+         prevNode = this.findTargetPrev(someTarget);
+         if (prevNode != null)
+         {
+            foundTarget = (E) prevNode.getNodeNext().getNodeData();
+            prevNode.next = prevNode.next.getNodeNext();
+            this.length--;
+         }   
+      }
+      return foundTarget;
+   }
+
+   /**
+    * Accessor function to find the previous Node in the linkedList where the 
+    * 1st match of the target is found. If the target matched the entry at the 
+    * head of the list this function will return null - making it look like the
+    * target was not found. It is the responsibility of the caller to check to 
+    * see if the head matches the target.
+    * @param someTarget The target we are looking for.
+    * @return The prevNode that is pointing to the target Node.
+    */
+   private ListNode<E> findTargetPrev(E someTarget)
+   {
+      ListNode curNode = this.head;
+      ListNode prevNode = null;
+      while (curNode != null)
+      {
+         if (someTarget.equals(curNode.getNodeData()))
+         {
+            return prevNode;
```

Comment:
After we find the element I think it would be better to do an additional check. if prevNode != null, return prevNode, else return curNode. This way it can find the target no matter what and the caller doesn't have to handle the precondition.